src/api/initialDataTypes.js

```diff
@@ -19,7 +19,30 @@ import type {
 export type InitialDataBase = $ReadOnly<{|
   last_event_id: number,
   msg: string,
-  queue_id: number,
+  queue_id: string,
+
+  // The feature level and server version, below, are included
+  // unconditionally as of 3.0 (zulip/zulip@2c6313019), which is why we put
+  // them in `InitialDataBase`. To get them from servers older than that, we
+  // have to either
+  // - pass `zulip_version` in `event_types`, or
+  // - not specify `event_types`.
+  // We've been doing the latter for a long time, so we're fine.
+
+  /**
+   * Added in server version 2.2, feature level 1.
+   * Same meaning as in the server_settings response:
+   * https://zulip.com/api/get-server-settings. See also the comment above.
+   */
+  zulip_feature_level?: number,
+
+  /**
+   * Should be present as far back as 1.6.0 (zulip/zulip@d9b10727f). Same
+   * meaning as in the server_settings response:
+   * https://zulip.com/api/get-server-settings. See also the comment above
+   * `zulip_feature_level`, above.
+   */
+  zulip_version: string,
```

Comment:
The commit message is just

    register: Note that server feature level is also in /register response.

but this `zulip_version` seems like the more interesting change.

Reply:
Hmm, also it looks like as of that early commit zulip/zulip@d9b10727f, this is conditional in exactly the same way that both this and `zulip_feature_level` are conditional until the later change zulip/zulip@2c6313019.

Maybe we should just send `zulip_version` in our `fetch_event_types`, so that this property is unconditionally present? With a `TODO(server-3.0)` comment, to help us remove it in the future.

Reply:
> Maybe we should just send `zulip_version` in our `fetch_event_types`, so that this property is unconditionally present? With a `TODO(server-3.0)` comment, to help us remove it in the future.

This is what I'd thought we had to do, and what made it into my earlier [post](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/connection.20failed/near/1248170) on CZO.

But looking closer, it's actually controlled by `event_types`, not `fetch_event_types`. So we should get the server version and (from servers that have one) the server feature level unconditionally, without having to do anything but keep passing `null` for `event_types`.